pkg/segment/query/processor/queryprocessor.go

```diff
@@ -125,7 +126,12 @@ func NewQueryProcessor(firstAgg *structs.QueryAggregators, queryInfo *query.Quer
 		dataProcessors[i].streams = append(dataProcessors[i].streams, NewCachedStream(dataProcessors[i-1]))
 	}
 
-	var lastStreamer streamer = searcher
+	if hook := hooks.GlobalHooks.GetDistributedStreamsHook; hook != nil {
+		result := hook(dataProcessors, searcher, queryInfo)
+		dataProcessors = result.([]*DataProcessor)
```

Comment:
nit: use `dataProcessors, ok = result.([]*DataProcessor)` and log an error if `!ok` so we avoid panicking